pynxtools_stm/reader.py

```diff
@@ -347,12 +250,33 @@ def read(
         # Get callable object that has parser inside
         parser = Spm().get_appropriate_parser(eln_dict)
         parser(template, data_file, config_dict, eln_dict)
+        set_default_attr_in_group(template)
 
-        set_default_for_each_group(template)
         manually_filter_data_type(template)
+        # for key, val in template.items():
+        #     if val is not None:
+        #         filled_template[key] = val
+
         for key, val in template.items():
-            if val is not None:
+            if isinstance(val, np.ndarray):
                 filled_template[key] = val
+            elif val not in (None, ""):
+                try:
+                    # Handle numpy array, list, tuple, etc.
+                    if str(val) not in (
```

Comment:
AFAICT, this is a repetition of the same (or similar) functionality in the `helpers.py/to_intended_t` function. Why not use that function here?

Reply:
Removed!